doc/rados/operations/placement-groups.rst

```diff
@@ -390,172 +407,162 @@ group will store objects on two OSDs, as shown below.
-there is no degradation of any object and it has no impact on the
-durability of the data contained in the Cluster.
+In such a cluster, the number of PGs has almost no effect on data durability.
+Whether there are 128 PGs per OSD or 8192 PGs per OSD, the recovery will be no
+slower or faster.
+
+However, an increase in the number of OSDs can increase the speed of recovery.
+Suppose our Ceph cluster is expanded from 10 OSDs to 20 OSDs.  Each OSD now
+participates in only ~75 PGs rather than ~150 PGs. All 19 remaining OSDs will
+still be required to replicate the same number of objects in order to recover.
+But instead of there being only 10 OSDs that have to copy ~100 GB each, there
+are now 20 OSDs that have to copy only 50 GB each. If the network had
+previously been a bottleneck, recovery now happens twice as fast.
+
+Similarly, suppose that our cluster grows to 40 OSDs. Each OSD will host only
+~38 PGs. And if an OSD dies, recovery will take place faster than before unless
+it is blocked by another bottleneck. Now, however, suppose that our cluster
+grows to 200 OSDs. Each OSD will host only ~7 PGs. And if an OSD dies, recovery
+will happen across at most ~21 (7 * 3) OSDs associated with these PGs. This
+means that recovery will take longer than when there were only 40 OSDs. For
+this reason, the number of PGs should be increased.
+
+No matter how brief the recovery time is, there is always a chance that an
+additional OSD will fail while recovery is in progress.  Consider the cluster
+with 10 OSDs described above: if any of the OSDs fail, then ~17 (approximately
+150 divided by 9) PGs will have only one remaining copy. And if any of the 8
+remaining OSDs fail, then 2 (approximately 17 divided by 8) PGs are likely to
+lose their remaining objects.
+
+When the number of OSDs in the cluster increases to 20 OSDs, the number of PGs
+that would be damaged by the loss of three OSDs significantly decreases. The
+loss of a second OSD degrades only ~4 (approximately 75 divided by 19) PGs
+rather than ~17 PGs, and the loss of a third OSD results in data loss only if
+it is one of the 4 OSDs that contains the remaining copy. This means --
+assuming that the probability of losing one OSD during recovery is 0.0001% --
+that the probability of data loss when three OSDs are lost is ~17 * 10 *
+0.0001% in the cluster with 10 OSDs, and only ~4 * 20 * 0.0001% in the cluster
+with 20 OSDs.
+
+In summary, the greater the number of OSDs, the faster the recovery and the
+lower the risk of permanently losing a PG due to cascading failures. As far as
+data durability is concerned, in a cluster with fewer than 50 OSDs, it doesn't
+much matter whether there are 512 or 4096 PGs.
+
+.. note::  It can take a long time for an OSD that has been recently added to
+   the cluster to be populated with the PGs that were assigned to it.  However,
+   no object degradation or impact on data durability will result from the
+   slowness of this process.
```

Comment:
... since Ceph populates data into the new PGs before removing it from the old.

Reply:
Accepted.